### PR TITLE
Converting crabs to basic mobs

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Exactly what it reads on the tin. As a bonus, they will flee from attacking targets, hunt tiny critters (crabs are now small-sized) and actually move sideways (it's an element that covers both client and basic movement)

## Why It's Good For The Game
Another simple to basic mob refactor.

## Changelog

:cl:
refactor: Crabs refactored into basic mobs. They now hunt tiny critters and flee from attackers.
fix: Fixed crabs not crab-walking.
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/basic_subtrees/simple_find_nearest_target_to_flee.dm [hunk 1]
```diff
@@ -9,9 +9,13 @@
 
 /// Find the nearest thing on our list of 'things which have done damage to me' and set it as the flee target
 /datum/ai_planning_subtree/find_nearest_thing_which_attacked_me_to_flee
+	var/targetting_key = BB_TARGETTING_DATUM
```
Comment (san7890) [inline code/datums/ai/basic_mobs/basic_subtrees/simple_find_nearest_target_to_flee.dm:12]:
```suggestion
	var/targeting_key = BB_TARGETTING_DATUM
```

spelling nit

File: code/datums/memory/_memory.dm [hunk 2]
```diff
@@ -268,7 +268,7 @@
 		/mob/living/basic/statue,
 		/mob/living/basic/stickman,
 		/mob/living/basic/stickman/dog,
-		/mob/living/simple_animal/crab,
+		/mob/living/basic/crab,
```
Comment (san7890) [inline code/datums/memory/_memory.dm:271]: alphabetize please

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -1414,7 +1414,7 @@
 				/mob/living/basic/statue,
 				/mob/living/basic/stickman,
 				/mob/living/basic/stickman/dog,
-				/mob/living/simple_animal/crab,
+				/mob/living/basic/crab,
```
Comment (san7890) [inline code/modules/mob/living/living.dm:1417]: alphabetize

File: code/modules/research/experimentor.dm [hunk 4]
```diff
@@ -647,7 +647,7 @@
 		/mob/living/basic/pet/dog/corgi,
 		/mob/living/basic/pet/dog/pug,
 		/mob/living/basic/pet/fox,
-		/mob/living/simple_animal/crab,
+		/mob/living/basic/crab,
```
Comment (san7890) [inline code/modules/research/experimentor.dm:650]: alphabetize please

**Review by san7890 — CHANGES_REQUESTED** [review]
i was looking into this but got completely shrekt looking into forcing sideways movement because I kept running into snags, thanks for cracking the issue

File: code/datums/elements/sideway_movement.dm [hunk 5]
```diff
@@ -17,8 +17,11 @@
...
 
+///retain the old dir unless walking straight ahead or backward, in which case rotate the dir left or right by a 90°
 /datum/element/sideway_movement/proc/on_move(atom/movable/source, direction, old_dir)
 	if(!source.set_dir_on_move)
 		return
-	if(direction == old_dir) //This means more or less we're walking straight. we don't want that.
-		source.setDir(angle2dir(dir2angle(direction) + pick(90, -90)))
+	var/new_dir = old_dir
+	if(direction == old_dir || direction == REVERSE_DIR(old_dir))
+		new_dir = angle2dir(dir2angle(direction) + pick(90, -90))
```
Comment (san7890) [inline code/datums/elements/sideway_movement.dm:26]: yeah this shit was exactly the reason why i gave up lol (no change)

Comment (Ghommie) [reply code/datums/elements/sideway_movement.dm:26]: come to think of it, I could have simply done:
```suggestion
		new_dir = turn(direction, pick(90, -90))
```
That was dumb of me.

Comment (san7890) [reply code/datums/elements/sideway_movement.dm:26]: didn't even know that was a proc lol. never too late to make another PR

**Pull request merged**